core/variant_parser.cpp

```diff
@@ -744,7 +744,7 @@ Error VariantParser::parse_value(Token &token, Variant &value, Stream *p_stream,
 						return err;
 
 					if (token.type == TK_PARENTHESIS_CLOSE) {
-
+						value = REF(obj->cast_to<Reference>());
```

Comment:
We could do this instead:

```cpp
if (ClassDB::is_parent_class(obj->get_class_name(), "Reference")) {
    value = REF(obj->cast_to<Reference>());		
} else {
    value = obj;
}
```

I've seen both patterns used in this code base.  When checking if something is a Node, ive seen ClassDB used, but when checking if something is a Reference, ive seen cast_to used.  But shouldn't the "cast_to" approach be much faster?  "is_parent_class" needs to walk the class dependency tree and do a bunch of string comparisons.

Am I wrong or did you have a different idea?